nixos/modules/services/continuous-integration/gitlab-runner-multi.nix

```diff
@@ -0,0 +1,231 @@
+          "gitlab-runner register"
+          "--non-interactive"
+          "--name ${name}"
+          "--executor ${service.executor}"
+          "--limit ${toString service.limit}"
+          "--request-concurrency ${toString service.requestConcurrency}"
+        ] ++ optional (service.dockerImage != null)
+          "--docker-image ${service.dockerImage}"
+          ++ optional service.dockerDisableCache
+          "--docker-disable-cache"
+          ++ optional (service.preCloneScript != null)
+          "--pre-clone-script ${service.preCloneScript}"
+          ++ optional (service.preBuildScript != null)
+          "--pre-build-script ${service.preBuildScript}"
+          ++ optional (service.postBuildScript != null)
+          "--post-build-script ${service.postBuildScript}"
+          ++ optional (service.tagList != [ ])
+          "--tag-list ${concatStringsSep "," service.tagList}"
+          ++ optional service.runUntagged
+          "--run-untagged"
+          ++ map (e: "--env ${escapeShellArg e}") service.environmentVariables
+          ++ map (v: "--docker-volumes ${escapeShellArg v}") service.dockerVolumes
+        ))} && sleep 1
+      fi
+    '') hashedServices)}
+
+    # unregister old services
+    for NAME in $(echo "$OLD_SERVICES")
+    do
+      [ ! -z "$NAME" ] && gitlab-runner unregister \
+        --name "$NAME" && sleep 1
+    done
+  '';
+in
+{
+  options.services.gitlab-runner-multi = {
+    enable = mkEnableOption "GiLab Runner Multiple services";
+    gracefulTermination = mkOption {
+      default = false;
+      type = types.bool;
+      description = ''
+        Finish all remaining jobs before stopping, restarting or reconfiguring.
+        If not set gitlab-runner will stop immediatly without waiting for jobs to finish,
+        which will lead to failed builds.
+      '';
+    };
+    gracefulTimeout = mkOption {
+      default = "infinity";
+      type = types.str;
+      example = "5min 20s";
+      description = ''
+        Time to wait until a graceful shutdown is turned into a forceful one.
+      '';
+    };
+    extraPackages = mkOption {
+      default = [ ];
+      type = types.listOf types.package;
+      description = ''
+        Extra packages to add to PATH for the gitlab-runner process.
+      '';
+    };
+    services = mkOption {
+      description = "GitLab Runner services";
+      default = { };
+      type = types.attrsOf (types.submodule {
+        options = {
+          registrationConfigFile = mkOption {
+            description = ''
+              Configuration file used got gitlab-runner registration.
+              It is a list of environment variables.
+              A list of all supported environment variables can be found
+              in
+                 gitlab-runner register --help
+              One that you probably want to set is
+              CI_SERVER_URL=<CI server URL>
+              REGISTRATION_TOKEN=<registration secret>
```

Comment:
How can I use this module without adding secrets to the nix-store?

Reply:
> How can I use this module without adding secrets to the nix-store?

Just set `registrationConfigFile` to some path.